core/src/main/java/nl/nn/adapterframework/soap/WsdlGeneratorUtils.java

```diff
@@ -123,16 +124,16 @@ static String validUri(String uri) {
         return uri == null ? null : uri.replaceAll(" ", "_");
     }
 
-    //check if the adapter has WebServiceListener with an inputValidator 
+	//check if the adapter has WebServiceListener with an inputValidator 
 	public static boolean canHaveWsdl(Adapter adapter) {
 		IValidator inputValidator = adapter.getPipeLine().getInputValidator();
-		boolean haveWebServiceListener = false;
-		for (IListener listener : WsdlGeneratorUtils.getListeners(adapter)) {
+		boolean hasWebServiceListener = false;
+		for (IListener<?> listener : WsdlGeneratorUtils.getListeners(adapter)) {
 			if(listener instanceof WebServiceListener) {
-				haveWebServiceListener = true;
+				hasWebServiceListener = true;
 			}
 		}
-		return inputValidator != null && haveWebServiceListener;
+		return inputValidator != null && ((inputValidator instanceof ApiWsdlXmlValidator) || hasWebServiceListener);
```

Comment:
Moet het zo specifiek deze validator class zijn?

Reply:
Ja ga ik in een andere PR oplossen